code/obj/flock/structure/gnesisturret.dm

```diff
@@ -120,25 +118,26 @@
 					logTheThing(LOG_COMBAT, src, "Flock gnesis turret at [log_loc(src)] belonging to flock [src.flock?.name] fires at [constructTarget(src.target)].")
 					SPAWN(0)
 						for(var/i in 1 to src.current_projectile.shot_number) //loop animation until finished
-							muzzle_flash_any(src, 0, "muzzle_flash")
+							muzzle_flash_any(src, get_angle(src, src.target), "muzzle_flash")
 							sleep(src.current_projectile.shot_delay)
 					shoot_projectile_ST_pixel_spread(src, current_projectile, target, 0, 0 , spread)
 
 	proc/seek_target()
 		var/list/target_list = list()
-		for (var/mob/living/C in range(src.range,src.loc))
+		for (var/mob/living/C in hearers(src.range,src.loc))
```

Comment:
Would think to change to viewers for parity with other Flock stuff, and Flockdrones at least being able to see cloaked stuff, if that extends to structures?

Reply:
done. also I made it `oviewers` for a slight efficiency tweak over `viewers`, since nothing is gonna be standing on top of the turret.